src/cocotb/_scheduler.py

```diff
@@ -172,61 +173,94 @@ def thread_wait(self):
+        Triggers are then `primed` (with :meth:`~cocotb.triggers.Trigger._prime`)
+        with a `react` function (:meth:`_sim_react` or :meth:`_react)
+        so as to wake up Tasks waiting for that Trigger to `fire` (when the event encoded by the Trigger occurs).
+        This is accomplished by :meth:`_resume_task_upon`.
+        :meth:`_resume_task_upon` also associates the Trigger with the Task waiting on it to fire by adding them to the :attr:`_triggers2tasks` map.
+        If instead, a Task finishes, :meth:`_schedule` will cause the :class:`~cocotb.triggers.Join` trigger to fire.
+        Once a Trigger fires it calls the react function which queues all Tasks waiting for that Trigger to fire.
+        Then the process repeats.
+
+        When a Task is cancelled (:meth:`_unschedule`), it is removed from the Task queue if it is currently queued.
+        Also, the Task and Trigger are deassociated in the :attr:`_triggers2tasks` map.
+        If the cancelled Task is the last Task waiting on a Trigger, that Trigger is `unprimed` to prevent it from firing.
+
+    Simulator Phase:
+        All GPITriggers (triggers that are fired by the simulator) go through :meth:`_sim_react`
+        which looks at the fired GPITriggers to determine and track the current simulator phase cocotb is executing in.
+
+        Normal phase:
+            Corresponds to all non-ReadWrite and non-ReadOnly phases.
+            Any writes are cached for the next ReadWrite phase and do not happen immediately.
+            Scheduling :class:`~cocotb.triggers.ReadWrite` and :class:`~cocotb.triggers.ReadOnly` are valid.
+
+        ReadWrite phase:
+            Corresponds to ``cbReadWriteSynch`` (VPI) or ``vhpiCbRepLastKnownDeltaCycle`` (VHPI).
+            At the start of scheduling in this phase we play back all the *previously* cached write updates.
+            Any writes are cached for the next ReadWrite phase and do not happen immediately.
+            Scheduling :class:`~cocotb.triggers.ReadWrite` and :class:`~cocotb.triggers.ReadOnly` are valid.
+            One caveat is that scheduling a :class:`~cocotb.triggers.ReadWrite` while in this phase may not be valid.
+            If there were no writes applied at the beginning of this phase, there will be no more events in this time step,
+            and there will not be another ReadWrite phase in this time step.
+            Simulators generally handle this caveat gracefully by leaving you in the ReadWrite phase of the next time step.
+
+        ReadOnly phase
+            Corresponds to ``cbReadOnlySynch`` (VPI) or ``vhpiCbRepEndOfTimeStep`` (VHPI).
+            In this state we are not allowed to perform writes.
+            Scheduling :class:`~cocotb.triggers.ReadWrite` and :class:`~cocotb.triggers.ReadOnly` are *not* valid.
+
+    Caveats and Special Cases:
+        Tests are treated specially in the scheduler.
```

Comment:
```suggestion
        The scheduler treats Tests specially.
```
(to have the uppercase "Tests")